### PR TITLE
Add graphite query router support

### PR DESCRIPTION
**Pull request by shreyassrivatsan**
**What this PR does / why we need it**:
This change adds query routers for graphite endpoints, allowing us
to add additional logic to the graphite endpoints.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:
```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:
```documentation-note
NONE
```

### CODE REVIEW

File: src/query/api/v1/httpd/handler_test.go [hunk 1]
```diff
@@ -290,6 +293,35 @@ func TestHealthGet(t *testing.T) {
...
+			res := httptest.NewRecorder()
+			ctrl := gomock.NewController(t)
+			storage, _ := m3.NewStorageAndSession(t, ctrl)
+
+			h, err := setupHandler(storage)
+			require.NoError(t, err, "unable to setup handler")
+			err = h.RegisterRoutes()
+			require.NoError(t, err)
+			h.Router().ServeHTTP(res, req)
+			require.Equal(t, http.StatusBadRequest, res.Code, "Empty request")
```
Comment (wesleyk) [inline src/query/api/v1/httpd/handler_test.go:320]: would this be a 404 if things weren't setup properly?

File: src/query/api/v1/httpd/graphite_router_test.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		called++
+	}
+
+	router := NewGraphiteFindRouter()
+	router.Setup(options.GraphiteFindRouterOptions{
+		FindHandler: findHandler,
+	})
+	rr := httptest.NewRecorder()
+
+	req, err := http.NewRequestWithContext(context.Background(), "GET", "/render?target=sum(metric)", nil)
```
Comment (wesleyk) [inline src/query/api/v1/httpd/graphite_router_test.go:65]: do these URLs actually matter? Confused that "/render" is passed in for the "find handler" test case

**Review by wesleyk — APPROVED** [review]
LGTM

**Review by ryanhall07 — APPROVED** [review]

File: src/query/api/v1/httpd/graphite_router_test.go (hunk 2 above)
Comment (shreyassrivatsan) [reply src/query/api/v1/httpd/graphite_router_test.go:65]: in this test dont really matter as we are directly calling the serveHttp method.

**Pull request merged**